### DISCUSSION TITLE
How to download the binary NAudio.dll?

### DISCUSSION THREAD

**Discussion by trojannemo** [naudio/NAudio#1074] [Q&A] [answered]
In the main Github page it says "click here to download the binary" but it takes you to the release page where it's only the source code. I tried loading the source code into my Visual Studio 2022/Windows 11 environment and it came up with 159 errors that i really don't want to spend time fixing. My current binary version of NAudio is 1.7.2.19 which the binary says is from 2014. I'm having compatibility issues with NET 6 and x64 so I was hoping to get a newer version of NAudio and that would work with my current needs. But I can't find it anywhere. The Github also says the latest version is 2.1.0 but this website says it is 2.2.1. Which is correct? 

https://libraries.io/nuget/NAudio/2.2.1

I would greatly appreciate your assistance in getting a compiled version of NAudio 2.1+, preferably one that will work with a x64 environment. Thanks.

**Comment by markheath** [comment]
The recommended way to consume NAudio is to use NuGet.
https://www.nuget.org/packages/NAudio

**Reply by trojannemo** [reply]
I wish there was a binary. I've spent hours trying to get NuGet to work on Visual Studio 2022. It keeps throwing an error. I've tried browsing nuget.org for naudio.dll and it gives an error. I've tried using the command line from within Visual Studio and it fails. Both inside the NAudio solution and from another solution (the one I want to use it in). All attempts lead to errors and nothing gets downloaded or installed. For now I found a binary that someone else included in their project dated January of 2020. Much newer than my 2014 one. But of course it's still oudated compared to 2.2.1 from 2023.

**Reply by markheath** [reply, marked as answer]
What is the error? Just about every .NET library uses NuGet as a distribution mechanism, it's very easy to use both in Visual Studio and from the .NET command line. 

Also you can download the NAudio package directly from the NuGet website. It will be a `.nupkg` file but it's just a zip file so if you rename it to `.zip` you can open it and find all the DLLs (there are several) in the `lib` folder.

**Reply by hirschi81** [reply]
It was also not possible for me to install the package for .NET 4.7.2. The Package Manager got me the following error message:
_"Could not install package 'NAudio.Asio 2.2.1'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.7.2', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author."_
This is strange, because ist says here: https://www.nuget.org/packages/NAudio#supportedframeworks-body-tab
that "net472" is "Included target framework(s) (in package)".

Thanks for the hint with unzipping the package. That helped with NAudio. But I then needed the NAudio.WASAPI package, but this is not available for net472, is it?
So I made a minor Update to 1.10.0
